src/main/java/com/google/sps/app/servlets/LogIn.java

```diff
@@ -34,11 +30,12 @@ public class LogIn extends HttpServlet {
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String username = request.getParameter(USER_NAME_PARAM).trim();
-
-        // if(!Validator.validUserName(username)) {
-        //     response.getWriter().println("Invalid Username: Improper characters. Please use letters and numbers only.");
-        //     return;
-        // }
+        String error = Validator.validateUserName(username);
+        if(error != "") {
+            error = error == "length" ? USER_NAME_LENGTH : IMPROPER_CHARACTERS;
```

Comment:
error.isEqualTo("length")

"In String, the == operator is used to comparing the reference of the given strings, depending on if they are referring to the same objects. When you compare two strings using == operator, it will return true if the string variables are pointing toward the same java object. Otherwise, it will return false "

Reply:
Good catch! I totally missed that